### PR TITLE
Refactor code to remove usage of Jsoup and Rssparser

### PR DESCRIPTION
**Pull request by maskaravivek**
## Title (required)

As part of #1863 this PR removes the usage of `Jsoup` and `Rssparser`

## Description (required)

Refactored the following: 
- Author name in Explore
- Started using MediaWikiAPI to fetch picture of the day. 

## Tests performed (required)

Tested on ProdDebug

## Screenshots showing what changed (optional)
 
explore
potd

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/utils/DateUtils.java [hunk 1]
```diff
@@ -33,4 +35,10 @@ public static String getTimeAgo(Date currDate, Date itemDate) {
             return "0-seconds";
         }
     }
+
+    public static String getCurrentDate() {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.US);
```
Comment (albendz) [inline app/src/main/java/fr/free/nrw/commons/utils/DateUtils.java]: Should this Locale always be US?

Comment (maskaravivek) [reply app/src/main/java/fr/free/nrw/commons/utils/DateUtils.java]: Yes, as the template string (eg. `Template:Potd/2015-08-02`) needs to be in this [format](https://www.mediawiki.org/wiki/API:Showing_interesting_content) irrespective of what locale the user is in.

**Comment by maskaravivek** [comment]
@albendz Thanks for reviewing the code. I have tested the changes manually. 

According to the [contribution guidelines](https://github.com/commons-app/apps-android-commons/wiki/Contributing-Guidelines#write-tests-for-your-code-if-possible), writing tests is recommended but not mandated.

**Comment by albendz** [comment]
The guidelines state writing unit tests if possible. It's definitely possible in this case. I would not request unit tests for any of the UI components since that is arguably not possible.

@misaochan Could you provide some clarity on whether unit tests would be appropriate here?

Apologies as I'm new to reviewing and have been trained to push for more unit tests where they make sense.

**Comment by maskaravivek** [comment]
@albendz I am on the same page about having as much coverage as possible in the codebase. It's just that I picked up refactoring this piece voluntarily and it might be discouraging for me(or other contributors) if I am supposed to add unit tests for the code I write. 

@misaochan @neslihanturan @nicolas-raoul It would be great to have your views on the same. Maybe we can continue the discussion on #1883.

**Comment by misaochan** [comment]
Unit tests would be great, but the main problem is the additional time needed. So that is why we currently don't "require" them in the sense that we would decline PRs for the lack of them, especially if the PR would reduce bugs/code issues as this one currently would. 

I am open to changing the policy depending on the results of the discussion at #1883 , but generally policy changes don't apply retroactively. :) @albendz So I would just review the code, perform manual testing, and then merge if satisfactory.

**Comment by albendz** [comment]
Alright, I will do manual testing shortly

**Comment by albendz** [comment]
The UI I'm seeing doesn't look like what is in your screenshots. Photos attached.
screenshot_1536360679
capture
 I'm also not seeing the photo of the day widget available in my homescreen widget options.

**Comment by maskaravivek** [comment]
@albendz The change has been made in Explore :)

**Comment by albendz** [comment]
How do I use Explore? I can't find it in the menu.

**Comment by maskaravivek** [comment]
@albendz Make sure that you are on the correct branch and have the latest changes. You can use these commands to pull the changes. 

```
git checkout -b maskaravivek-refactoring master
git pull https://github.com/maskaravivek/apps-android-commons.git refactoring
```

**Comment by neslihanturan** [comment]
Passes from my tests @maskaravivek , you can merge after conflicts are fixed, thanks!

**Comment by maskaravivek** [comment]
Thanks for the review guys. Going ahead and merging it. :)

**Pull request merged**